includes/core-changes/aspnetcore/5.0/http-badhttprequestexception-obsolete.md

```diff
@@ -0,0 +1,48 @@
+### HTTP: Kestrel and IIS BadHttpRequestException types marked obsolete and replaced
+
+`Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException` and `Microsoft.AspNetCore.Server.IIS.BadHttpRequestException` have been marked obsolete and changed to derive from `Microsoft.AspNetCore.Http.BadHttpRequestException`. The Kestrel and IIS servers still throw their old exception types for backwards compatibility. The obsolete types will be removed in a future release.
+
+For discussion, see [dotnet/aspnetcore#20614](https://github.com/dotnet/aspnetcore/issues/20614).
+
+#### Version introduced
+
+5.0
+
+#### Old behavior
+
+`Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException` and `Microsoft.AspNetCore.Server.IIS.BadHttpRequestException` derived from <xref:System.IO.IOException?displayProperty=nameWithType>.
+
+#### New behavior
+
+`Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException` and `Microsoft.AspNetCore.Server.IIS.BadHttpRequestException` are obsolete. The types also derive from `Microsoft.AspNetCore.Http.BadHttpRequestException`, which derives from `System.IO.IOException`.
+
+#### Reason for change
+
+The change was made to:
+
+* Consolidate duplicate types.
+* Unify behavior across server implementations.
+
+An app can now catch the base exception `Microsoft.AspNetCore.Http.BadHttpRequestException` when using either Kestrel or IIS.
+
+#### Recommended action
+
+Replace usages of `Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException` and `Microsoft.AspNetCore.Server.IIS.BadHttpRequestException` with `Microsoft.AspNetCore.Http.BadHttpRequestException`.
```

Comment:
```suggestion
Replace usages of `Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException` and `Microsoft.AspNetCore.Server.IIS.BadHttpRequestException` with <xref:Microsoft.AspNetCore.Http.BadHttpRequestException?displayProperty=nameWithType>.
```

Reply:
I wanted to do this, but it's a 5.0 API that hasn't been published yet. That's likely to happen within the next few weeks.